File: 02_Two_Pointers/C++ Code/02_392. Is Subsequence.cpp
// Problem Link -> https://leetcode.com/problems/is-subsequence/description/

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;
        while(i < s.length() && j < t.length()){
            if(s[i] == t[j]) i++;
            j++;
        }

        if(i == s.length()) return true;
        return false;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(1)

File: 02_Two_Pointers/Python Code/02_392. Is Subsequence.py
# Problem Link -> https://leetcode.com/problems/is-subsequence/description/

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ptr1, ptr2 = 0, 0
        m = len(s)
        n = len(t)

        while ptr1 < m and ptr2 < n:
            if s[ptr1] == t[ptr2]:
                ptr1 += 1
            ptr2 += 1
        
        return True if ptr1 == m else False



# Time Complexity -> O(n)
# Space Complexity -> O(1)
